runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -1930,13 +1930,613 @@ TR::Register *J9::Power::TreeEvaluator::anewArrayEvaluator(TR::Node *node, TR::C
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, oolJumpLabel);
+   generateLabelInstruction(cg, TR::InstOpCode::b, node, oolFailLabel);
+
+   // we need to include post dependencies from the call node if they didn't copy the children nodes
+   // (in which case they should be handled inside the function), and they use registers different
+   // from the main line
+   // I don't know if this is possible at all, every time I tested the results are just all zeros
+   TR::Node *callNode = outlinedHelperCall->getCallNode();
+   int callUsesFirstChild = (callNode->getFirstChild() == node->getFirstChild()) &&
+                            (callNode->getFirstChild()->getRegister()) &&
+                            (callNode->getFirstChild()->getRegister() != dimsPtrReg);
+   int callUsesSecondChild = (callNode->getSecondChild() == node->getSecondChild()) &&
+                            (callNode->getSecondChild()->getRegister());
+   int callUsesThirdChild = (callNode->getThirdChild() == node->getThirdChild()) &&
+                            (callNode->getThirdChild()->getRegister()) &&
+                            (callNode->getThirdChild()->getRegister() != classReg);
+   int numDeps = 12 + callUsesFirstChild + callUsesSecondChild + callUsesThirdChild;
+
+   TR::RegisterDependencyConditions *dependencies =
+      new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, numDeps, cg->trMemory());
+   dependencies->addPostCondition(dimsPtrReg, TR::RealRegister::NoReg);
+   dependencies->getPostConditions()->getRegisterDependency(dependencies->getAddCursorForPost() - 1)->setExcludeGPR0();
+
+   dependencies->addPostCondition(classReg, TR::RealRegister::NoReg);
+   dependencies->getPostConditions()->getRegisterDependency(dependencies->getAddCursorForPost() - 1)->setExcludeGPR0();
+
+
+   dependencies->addPostCondition(targetReg, TR::RealRegister::NoReg);
+   dependencies->getPostConditions()->getRegisterDependency(dependencies->getAddCursorForPost() - 1)->setExcludeGPR0();
+
+   dependencies->addPostCondition(firstDimLenReg, TR::RealRegister::NoReg);
+   dependencies->addPostCondition(secondDimLenReg, TR::RealRegister::NoReg);
+
+   dependencies->addPostCondition(temp1Reg, TR::RealRegister::NoReg);
+   dependencies->getPostConditions()->getRegisterDependency(dependencies->getAddCursorForPost() - 1)->setExcludeGPR0();
+
+   dependencies->addPostCondition(temp2Reg, TR::RealRegister::NoReg);
+   dependencies->getPostConditions()->getRegisterDependency(dependencies->getAddCursorForPost() - 1)->setExcludeGPR0();
+   dependencies->addPostCondition(temp3Reg, TR::RealRegister::NoReg);
+   dependencies->getPostConditions()->getRegisterDependency(dependencies->getAddCursorForPost() - 1)->setExcludeGPR0();
+
+   dependencies->addPostCondition(subArraySizeReg, TR::RealRegister::NoReg);
+
+   dependencies->addPostCondition(vmThreadReg, TR::RealRegister::NoReg); // do we need this?
```

Comment:
for Power codegen, vmThreadReg is a locked register, i.e.  it doesn't participate in register assignment.  Just like the java stack pointer register, you don't need to put it on dependencyConditions.

Also, wondering if you really need these many different registers.  could some of them have distinct live-ranges and be re-used in the between ... thus reducing the number of registers being used?  have you checked this out?